packages/cloud_firestore/cloud_firestore_platform_interface/lib/src/method_channel/method_channel_field_value_factory.dart

```diff
@@ -26,12 +24,13 @@ class MethodChannelFieldValueFactory extends FieldValueFactoryPlatform {
       MethodChannelFieldValue(FieldValueType.delete, null);
 
   @override
-  MethodChannelFieldValue increment(num value) {
+  MethodChannelFieldValue? increment(num value) {
```

Comment:
should be non-nullable and throw if != int/double instead of `return null`, do `throw StateError('...')`